app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -2142,24 +2143,47 @@ class BrowserTabViewModel @Inject constructor(
     }
 
     private fun initializeViewStates() {
+        initializeDefaultViewStates()
+        viewModelScope.launch {
+            initializeViewStatesFromPersistedData()
+        }
+    }
+
+    private fun initializeDefaultViewStates() {
         globalLayoutState.value = Browser()
-        browserViewState.value = BrowserViewState().copy(
-            addToHomeVisible = addToHomeCapabilityDetector.isAddToHomeSupported(),
-            showAutofill = autofillStore.autofillAvailable
-        )
+        browserViewState.value = BrowserViewState()
         loadingViewState.value = LoadingViewState()
         autoCompleteViewState.value = AutoCompleteViewState()
-        omnibarViewState.value = OmnibarViewState(
-            showVoiceSearch = voiceSearchAvailability.shouldShowVoiceSearch()
-        )
+        omnibarViewState.value = OmnibarViewState()
         findInPageViewState.value = FindInPageViewState()
         ctaViewState.value = CtaViewState()
         privacyGradeViewState.value = PrivacyGradeViewState()
-        accessibilityViewState.value = AccessibilityViewState(
-            fontSize = accessibilitySettingsDataStore.fontSize,
-            forceZoom = accessibilitySettingsDataStore.forceZoom,
-            refreshWebView = false
-        )
+        accessibilityViewState.value = AccessibilityViewState()
+    }
+
+    private suspend fun initializeViewStatesFromPersistedData() {
+        withContext(dispatchers.io()) {
+            val addToHomeSupported = addToHomeCapabilityDetector.isAddToHomeSupported()
+            val showAutofill = autofillStore.autofillAvailable
+            val showVoiceSearch = voiceSearchAvailability.shouldShowVoiceSearch()
+            val fontSize = accessibilitySettingsDataStore.fontSize
+            val forceZoom = accessibilitySettingsDataStore.forceZoom
+
+            withContext(dispatchers.main()) {
+                browserViewState.value = currentBrowserViewState().copy(
+                    addToHomeVisible = addToHomeSupported,
+                    showAutofill = showAutofill
+                )
+                omnibarViewState.value = currentOmnibarViewState().copy(
+                    showVoiceSearch = showVoiceSearch
+                )
+                accessibilityViewState.value = currentAccessibilityViewState().copy(
```

Comment:
I don't think we need to initialize the `ViewState`. If we have a default value for the `ViewState`, initialization can happen inside the observer we create in `observeAccessibilitySettings`.

Reply:
Agreed.

I was trying to best replicate what was there before, including that initial initialization before `observeAccessibilitySettings`, but agree that should be unnecessary. Most likely unnecessary before too. Will remove.